dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -2372,8 +2367,10 @@ int ha_mcs_impl_rnd_next(uchar* buf, TABLE* table)
                 thd->lex->sql_command == SQLCOM_LOAD))
         return HA_ERR_END_OF_FILE;
 
-    if (((thd->lex)->sql_command == SQLCOM_UPDATE)  || ((thd->lex)->sql_command == SQLCOM_DELETE) ||
-            ((thd->lex)->sql_command == SQLCOM_DELETE_MULTI) || ((thd->lex)->sql_command == SQLCOM_UPDATE_MULTI))
+    if ((thd->lex->sql_command == SQLCOM_UPDATE) ||
```

Comment:
Is it possible to replace this repeating conditions with the bool func you created?

Reply:
Not possible in this case since the following `&&` goes together:
`((thd->lex->sql_command == SQLCOM_UPDATE_MULTI) && !isForeignTableUpdate(thd))`

Reply:
I think isUpdateStatement() can have a aForeignTableUpdate argument with a default value. isUpdateOrDeleteStatement() call will propagate isForeignTableUpdate() result into isUpdateStatement(). Feel free to ask in Slack.